src/NHibernate/Mapping/ByCode/Impl/BagMapper.cs

```diff
@@ -249,6 +249,16 @@ public void SqlInsert(string sql)
 			mapping.sqlinsert.Text = new[] {sql};
 		}
 
+		public void SqlInsert(string sql, SqlCheck sqlCheck)
+		{
+			if (mapping.SqlInsert == null)
+			{
+				mapping.sqlinsert = new HbmCustomSQL();
+			}
+			mapping.sqlinsert.Text = new[] {sql};
+			mapping.sqlinsert.check = sqlCheck.ToHbmSqlCheck();
+		}
```

Comment:
Why no `mapping.sqlinsert.checkSpecified = true;` here? Just forgotten from all sqlinsert in collections?

Reply:
Yes, probably forgot.

Reply:
Thanks. Fixed.